src/StepperFirmata2.cpp

```diff
@@ -0,0 +1,432 @@
+        float speed = decodeCustomFloat(argv[2], argv[3], argv[4], argv[5]);
+
+        if (stepper[deviceNum]) {
+          stepper[deviceNum]->setMaxSpeed(speed);
+        }
+
+      }
+
+      else if (stepCommand == MULTISTEPPER_CONFIG) {
+        
+        if (!group[deviceNum]) {
+          numGroups++;
+          group[deviceNum] = new MultiStepper();
+        }
+
+        for (byte i = index; i < argc; i++) {
+          byte stepperNumber = argv[i];
+          
+          if (stepper[stepperNumber]) {
+            groupStepperCount[deviceNum]++;
+            group[deviceNum]->addStepper(*stepper[stepperNumber]);
+          }
+        }
+
+        groupIsRunning[deviceNum] = false;
+      }
+
+      else if (stepCommand == MULTISTEPPER_TO) {
+
+        groupIsRunning[deviceNum] = true;
+        long positions[groupStepperCount[deviceNum]];
+        
+        for (byte i = 0, offset = 0; i < groupStepperCount[deviceNum]; i++) {
+          offset = index + (i * 5);
+          positions[i] = decode32BitSignedInteger(argv[offset], argv[offset+1], argv[offset+2], argv[offset+3], argv[offset+4]);
+        }
+
+        group[deviceNum]->moveTo(positions);
+        
+      }
+
+      else if (stepCommand == MULTISTEPPER_STOP) {
+
+        groupIsRunning[deviceNum] = false;
+        
+      }
+    }
+    return true;
+    
+  }
+  return false;
+}
+
+/*==============================================================================
+ * SETUP()
+ *============================================================================*/
+
+void StepperFirmata2::reset()
+{
+  for (byte i = 0; i < MAX_STEPPERS; i++) {
+    if (stepper[i]) {
+      free(stepper[i]);
+      stepper[i] = 0;
+    }
+  }
+  numSteppers = 0;
+  
+  for (byte i = 0; i < MAX_GROUPS; i++) {
+    if (group[i]) {
+      free(group[i]);
+      group[i] = 0;
+    }
+  }
+  numGroups = 0;
+}
+
+/*==============================================================================
+ * Helpers
+ *============================================================================*/
+
+float StepperFirmata2::decodeCustomFloat(byte arg1, byte arg2, byte arg3, byte arg4)
+{
+  long l4 = (long)arg4;
+  long significand = (long)arg1 | (long)arg2 << 7 | (long)arg3 << 14 | (l4 & 0x03) << 21;
+  float exponent = (float)(((l4 >> 2) & 0x0f) - 10);
+  bool sign = (bool)((l4 >> 6) & 0x01);
+  float result = (float)significand;
+
+  if (sign) {
+    result *= -1;
+  }
+
+  result = result * powf(10.0, exponent);
+
+  return result;
+}
+
+long StepperFirmata2::decode28BitUnsignedInteger(byte arg1, byte arg2, byte arg3, byte arg4)
+{
+  long result = (long)arg1 | (long)arg2 << 7 | (long)arg3 << 14 | (long)arg4 << 21;
+
+  return result;
+}
+
+long StepperFirmata2::decode32BitSignedInteger(byte arg1, byte arg2, byte arg3, byte arg4, byte arg5)
+{
+  long result = (long)arg1 | (long)arg2 << 7 | (long)arg3 << 14 | (long)arg4 << 21 | ((long)arg5 << 28) & 0x07;
```

Comment:
Add a set of parens around the last byte: `(((long)arg5 << 28) & 0x07)`. That will make a complier warning go away.